### PR TITLE
[android] Add compass drawable initialization in UiSettings

### LINKED ISSUE #8549: [android] Polygon test app example crashes when rotating screen
**Platform:** Android
**Mapbox SDK version:** `master` branch

### Steps to trigger behavior

 1. Open Polygon example
 2. Rotate screen
 3. App crashes

### Expected behavior
App doesn't crash

### Actual behavior
App is crashing

polygonrotationcrash

### Logcat
```
03-28 14:39:13.573 32317-32317/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.mapbox.mapboxsdk.testapp, PID: 32317
                                                   java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.drawable.Drawable.getIntrinsicWidth()' on a null object reference
                                                       at com.mapbox.mapboxsdk.maps.MapboxMapOptions.getBitmapFromDrawable(MapboxMapOptions.java:160)
                                                       at com.mapbox.mapboxsdk.maps.UiSettings.saveCompass(UiSettings.java:164)
                                                       at com.mapbox.mapboxsdk.maps.UiSettings.onSaveInstanceState(UiSettings.java:87)
                                                       at com.mapbox.mapboxsdk.maps.MapboxMap.onSaveInstanceState(MapboxMap.java:128)
                                                       at com.mapbox.mapboxsdk.maps.MapView.onSaveInstanceState(MapView.java:229)
                                                       at com.mapbox.mapboxsdk.testapp.activity.annotation.PolygonActivity.onSaveInstanceState(PolygonActivity.java:109)
                                                       at android.app.Activity.performSaveInstanceState(Activity.java:1298)
                                                       at android.app.Instrumentation.callActivityOnSaveInstanceState(Instrumentation.java:1289)
                                                       at android.app.ActivityThread.callCallActivityOnSaveInstanceState(ActivityThread.java:4088)
                                                       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4050)
                                                       at android.app.ActivityThread.-wrap15(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:148)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

### PR DESCRIPTION
**Pull request by Guardiola31337**
Fixes #8549 
- Added compass drawable initialization in `UiSettings`.

:eyes: @tobrun

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java [hunk 1]
```diff
@@ -217,7 +218,7 @@ public static MapboxMapOptions createFromAttributes(@NonNull Context context, @N
       Drawable compassDrawable = typedArray.getDrawable(
         R.styleable.mapbox_MapView_mapbox_uiCompassDrawable);
       if (compassDrawable == null) {
-        compassDrawable = ContextCompat.getDrawable(context, R.drawable.mapbox_compass_icon);
+        compassDrawable = ResourcesCompat.getDrawable(context.getResources(), R.drawable.mapbox_compass_icon, null);
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: what is the benefit of using `ResoucesCompat` above `ContextCompat`?

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: @tobrun 
It allows you to provide the `Theme` argument yourself.
Using `ContextCompat`:
> Starting in LOLLIPOP, the returned drawable will be styled for the specified Context's theme.
They're deprecating all methods in which you don't specify the `Theme`, so it seems the way to go 😉

Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: AFAIK ContextCompat isn't deprecated and will not be deprecated. It provides compatibility with LOLLIPOP in `ContextCompatApi21.java`. Feel free to link more information on this.

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: You're right. I messed `getDrawable` APIs. [`Resources#getDrawable()`](https://developer.android.com/reference/android/content/res/Resources.html#getDrawable%28int%29) was the one deprecated. Actually the way to go is `ContextCompat` as pointed out [here](https://plus.google.com/+BenjaminWeiss/posts/M1dYFaobyBM).
Funny thing is that `ResourcesCompat` isn't deprecated 😂 
What a mess! First world problems of having multiple classes doing the same...

**Pull request merged**